irohad/torii/cache/cache.hpp

```diff
@@ -0,0 +1,82 @@
+/**
+ * Copyright Soramitsu Co., Ltd. 2017 All Rights Reserved.
+ * http://soramitsu.co.jp
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef IROHA_CACHE_HPP
+#define IROHA_CACHE_HPP
+
+#include <endpoint.pb.h>
+#include <boost/optional.hpp>
+#include <list>
+#include <mutex>
+#include <string>
+
+namespace torii {
+  namespace cache {
+    /**
+     * Cache for torii responses.
+     * Internally it uses uses a map to cache tx statuses and linked list
+     * based index to remove oldest items when getIndexSizeHigh() is reached.
+     */
+    class ToriiResponseCache {
```

Comment:
This class possible may be used as a mocked object. Thus, there is need to write an interface for it. I guess NVI idiom may be applicable here.

Reply:
Also, as I understand design better to make the class template for applying any custom objects inside.

Reply:
Another issue about class design. Maybe better to write two constructors here. One with low and max size parameters and another that used defaults?